logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/E2EIClientProvider.kt

```diff
@@ -51,17 +51,21 @@ internal class EI2EIClientProviderImpl(
 
     private val mutex = Mutex()
 
-    override suspend fun getE2EIClient(clientId: ClientId?, isNewClient: Boolean): Either<CoreFailure, E2EIClient> = mutex.withLock {
+    override suspend fun getE2EIClient(clientId: ClientId?, isNewClient: Boolean): Either<E2EIFailure, E2EIClient> = mutex.withLock {
         withContext(dispatchers.io) {
             val currentClientId =
-                clientId ?: currentClientIdProvider().fold({ return@withContext Either.Left(it) }, { it })
+                clientId ?: currentClientIdProvider().fold({ return@withContext E2EIFailure.GettingE2EIClient(it).left() }, { it })
 
             return@withContext e2EIClient?.let {
                 Either.Right(it)
             } ?: run {
-                getSelfUserInfo().flatMap { selfUser ->
+                getSelfUserInfo().fold({
```

Comment:
you could use `mapLeft` `map` and `flatMap` here, the result is the same but code would be cleaner, just FYI :)

Reply:
adjusted the code, thanks for the input.